graylog2-web-interface/src/components/common/TypeAheadDataFilter.jsx

```diff
@@ -131,59 +162,63 @@ class TypeAheadDataFilter extends React.Component {
   };
 
   filterData = () => {
-    if (typeof this.props.filterData === 'function') {
-      return this.props.filterData(this.props.data);
+    const { filterData, data, onDataFiltered } = this.props;
+    if (typeof filterData === 'function') {
+      return filterData(data);
     }
 
-    const filteredData = this.props.data.filter((datum) => {
+    const filteredData = data.filter((datum) => {
       return this._matchFilters(datum) && this._matchStringSearch(datum);
     }, this);
 
-    this.props.onDataFiltered(filteredData);
+    onDataFiltered(filteredData);
+    return true;
```

Comment:
I don't understand why we return `true` in this function, could you please explain that?

Reply:
This is to fix linter issue : https://eslint.org/docs/rules/consistent-return

Reply:
Okay. Looking at where this function is used, I think in this case we only use a return before in order to not execute the rest of the function when a `filterData` function is given. I think it would be clearer to return with no value in the if case and remove this line.